<a href="https://colab.research.google.com/github/MANISH-KUMAR-CODES/Exploratory-Analysis-on-International-t20-/blob/main/Exploratory_Analysis_on_International_t20.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

##<B>Analysis ON International t20

###Lets us first import all the necessary libraries of python

In [3]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [4]:
International_T20_Data = pd.read_csv('/content/drive/MyDrive/Copy of International_T20_Data.csv')

In [5]:
#lets check whats inside our data
International_T20_Data

innings  meta.data_version  \
0     [{'1st innings': {'team': 'Australia', 'delive...                0.9   
1     [{'1st innings': {'team': 'Australia', 'delive...                0.9   
2     [{'1st innings': {'team': 'Australia', 'delive...                0.9   
3     [{'1st innings': {'team': 'Hong Kong', 'delive...                0.9   
4     [{'1st innings': {'team': 'Zimbabwe', 'deliver...                0.9   
...                                                 ...                ...   
1412  [{'1st innings': {'team': 'Sri Lanka', 'delive...                0.9   
1413  [{'1st innings': {'team': 'Bangladesh', 'deliv...                0.9   
1414  [{'1st innings': {'team': 'Netherlands', 'deli...                0.9   
1415  [{'1st innings': {'team': 'Australia', 'delive...                0.9   
1416  [{'1st innings': {'team': 'Sri Lanka', 'delive...                0.9   

     meta.created  meta.revision                    info.dates info.gender  \
0      2017-02-18              2  [datetime.date(2017, 2, 17)]        male   
1      2017-02-19              2  [datetime.date(2017, 2, 19)]        male   
2      2017-02-23              1  [datetime.date(2017, 2, 22)]        male   
3      2016-09-12              1   [datetime.date(2016, 9, 5)]        male   
4      2016-06-19              1  [datetime.date(2016, 6, 18)]        male   
...           ...            ...                           ...         ...   
1412   2016-03-05              2   [datetime.date(2016, 3, 4)]        male   
1413   2016-03-08              1   [datetime.date(2016, 3, 6)]        male   
1414   2016-02-03              1   [datetime.date(2016, 2, 3)]        male   
1415   2016-09-12              1   [datetime.date(2016, 9, 6)]        male   
1416   2016-09-12              1   [datetime.date(2016, 9, 9)]        male   

     info.match_type  info.outcome.by.wickets info.outcome.winner  info.overs  \
0                T20                      5.0           Sri Lanka          20   
1                T20                      2.0           Sri Lanka          20   
2                T20                      NaN           Australia          20   
3                T20                      NaN           Hong Kong          20   
4                T20                      NaN            Zimbabwe          20   
...              ...                      ...                 ...         ...   
1412             T20                      6.0            Pakistan          20   
1413             T20                      8.0               India          20   
1414             T20                      NaN         Netherlands          20   
1415             T20                      NaN           Australia          20   
1416             T20                      4.0           Australia          20   

      ... info.outcome.by.runs info.match_type_number info.neutral_venue  \
0     ...                  NaN                    NaN                NaN   
1     ...                  NaN                    NaN                NaN   
2     ...                 41.0                    NaN                NaN   
3     ...                 40.0                    NaN                NaN   
4     ...                  2.0                    NaN                NaN   
...   ...                  ...                    ...                ...   
1412  ...                  NaN                    NaN                1.0   
1413  ...                  NaN                    NaN                NaN   
1414  ...                 84.0                    NaN                NaN   
1415  ...                 85.0                    NaN                NaN   
1416  ...                  NaN                    NaN                NaN   

     info.outcome.method info.outcome.result info.outcome.eliminator  \
0                    NaN                 NaN                     NaN   
1                    NaN                 NaN                     NaN   
2                    NaN                 NaN                     NaN   


In [6]:
#top 5 rows
International_T20_Data.head(5)

innings  meta.data_version  \
0  [{'1st innings': {'team': 'Australia', 'delive...                0.9   
1  [{'1st innings': {'team': 'Australia', 'delive...                0.9   
2  [{'1st innings': {'team': 'Australia', 'delive...                0.9   
3  [{'1st innings': {'team': 'Hong Kong', 'delive...                0.9   
4  [{'1st innings': {'team': 'Zimbabwe', 'deliver...                0.9   

  meta.created  meta.revision                    info.dates info.gender  \
0   2017-02-18              2  [datetime.date(2017, 2, 17)]        male   
1   2017-02-19              2  [datetime.date(2017, 2, 19)]        male   
2   2017-02-23              1  [datetime.date(2017, 2, 22)]        male   
3   2016-09-12              1   [datetime.date(2016, 9, 5)]        male   
4   2016-06-19              1  [datetime.date(2016, 6, 18)]        male   

  info.match_type  info.outcome.by.wickets info.outcome.winner  info.overs  \
0             T20                      5.0           Sri Lanka          20   
1             T20                      2.0           Sri Lanka          20   
2             T20                      NaN           Australia          20   
3             T20                      NaN           Hong Kong          20   
4             T20                      NaN            Zimbabwe          20   

   ... info.outcome.by.runs info.match_type_number info.neutral_venue  \
0  ...                  NaN                    NaN                NaN   
1  ...                  NaN                    NaN                NaN   
2  ...                 41.0                    NaN                NaN   
3  ...                 40.0                    NaN                NaN   
4  ...                  2.0                    NaN                NaN   

  info.outcome.method info.outcome.result info.outcome.eliminator  \
0                 NaN                 NaN                     NaN   
1                 NaN                 NaN                     NaN   
2                 NaN                 NaN                     NaN   
3                 NaN                 NaN                     NaN   
4                 NaN                 NaN                     NaN   

  info.supersubs.New Zealand  info.supersubs.South Africa  info.bowl_out  \
0                        NaN                          NaN            NaN   
1                        NaN                          NaN            NaN   
2                        NaN                          NaN            NaN   
3                        NaN                          NaN            NaN   
4                        NaN                          NaN            NaN   

   info.outcome.bowl_out  
0                    NaN  
1                    NaN  
2                    NaN  
3                    NaN  
4                    NaN  

[5 rows x 27 columns]

In [7]:
#last 5 rows
International_T20_Data.tail(5)

innings  meta.data_version  \
1412  [{'1st innings': {'team': 'Sri Lanka', 'delive...                0.9   
1413  [{'1st innings': {'team': 'Bangladesh', 'deliv...                0.9   
1414  [{'1st innings': {'team': 'Netherlands', 'deli...                0.9   
1415  [{'1st innings': {'team': 'Australia', 'delive...                0.9   
1416  [{'1st innings': {'team': 'Sri Lanka', 'delive...                0.9   

     meta.created  meta.revision                   info.dates info.gender  \
1412   2016-03-05              2  [datetime.date(2016, 3, 4)]        male   
1413   2016-03-08              1  [datetime.date(2016, 3, 6)]        male   
1414   2016-02-03              1  [datetime.date(2016, 2, 3)]        male   
1415   2016-09-12              1  [datetime.date(2016, 9, 6)]        male   
1416   2016-09-12              1  [datetime.date(2016, 9, 9)]        male   

     info.match_type  info.outcome.by.wickets info.outcome.winner  info.overs  \
1412             T20                      6.0            Pakistan          20   
1413             T20                      8.0               India          20   
1414             T20                      NaN         Netherlands          20   
1415             T20                      NaN           Australia          20   
1416             T20                      4.0           Australia          20   

      ... info.outcome.by.runs info.match_type_number info.neutral_venue  \
1412  ...                  NaN                    NaN                1.0   
1413  ...                  NaN                    NaN                NaN   
1414  ...                 84.0                    NaN                NaN   
1415  ...                 85.0                    NaN                NaN   
1416  ...                  NaN                    NaN                NaN   

     info.outcome.method info.outcome.result info.outcome.eliminator  \
1412                 NaN                 NaN                     NaN   
1413                 NaN                 NaN                     NaN   
1414                 NaN                 NaN                     NaN   
1415                 NaN                 NaN                     NaN   
1416                 NaN                 NaN                     NaN   

     info.supersubs.New Zealand  info.supersubs.South Africa  info.bowl_out  \
1412                        NaN                          NaN            NaN   
1413                        NaN                          NaN            NaN   
1414                        NaN                          NaN            NaN   
1415                        NaN                          NaN            NaN   
1416                        NaN                          NaN            NaN   

      info.outcome.bowl_out  
1412                    NaN  
1413                    NaN  
1414                    NaN  
1415                    NaN  
1416                    NaN  

[5 rows x 27 columns]

In [8]:
#Checking info
International_T20_Data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1417 entries, 0 to 1416
Data columns (total 27 columns):
 #   Column                       Non-Null Count  Dtype  
---  ------                       --------------  -----  
 0   innings                      1417 non-null   object 
 1   meta.data_version            1417 non-null   float64
 2   meta.created                 1417 non-null   object 
 3   meta.revision                1417 non-null   int64  
 4   info.dates                   1417 non-null   object 
 5   info.gender                  1417 non-null   object 
 6   info.match_type              1417 non-null   object 
 7   info.outcome.by.wickets      651 non-null    float64
 8   info.outcome.winner          1372 non-null   object 
 9   info.overs                   1417 non-null   int64  
 10  info.player_of_match         1255 non-null   object 
 11  info.teams                   1417 non-null   object 
 12  info.toss.decision           1417 non-null   object 
 13  info.toss.winner  

In [9]:
#checking descriptive summary
International_T20_Data.describe().T

count        mean           std    min    25%  \
meta.data_version        1417.0    0.900000  5.108829e-15    0.9    0.9   
meta.revision            1417.0    1.370501  5.751977e-01    1.0    1.0   
info.outcome.by.wickets   651.0    6.476190  1.987802e+00    1.0    5.0   
info.overs               1417.0   20.063514  1.379399e+00   20.0   20.0   
info.outcome.by.runs      721.0   39.088766  3.338576e+01    1.0   13.0   
info.match_type_number    684.0  748.774854  1.900503e+02  380.0  627.0   
info.neutral_venue        328.0    1.000000  0.000000e+00    1.0    1.0   

                           50%     75%     max  
meta.data_version          0.9    0.90     0.9  
meta.revision              1.0    2.00     5.0  
info.outcome.by.wickets    7.0    8.00    10.0  
info.overs                20.0   20.00    50.0  
info.outcome.by.runs      30.0   56.00   249.0  
info.match_type_number   746.5  883.25  1104.0  
info.neutral_venue         1.0    1.00     1.0

In [10]:
#lets us define a function to replace column names
def replace_name(dataframe):
    dataframe.astype(str)
    dataframe_new_column=[]
    for i in dataframe.columns:
       if  'meta.' in i:
          i = i.replace('meta.','')
          dataframe_new_column.append(i)
       else: 
             'info.' in i
             i = i.replace('info.','')
             dataframe_new_column.append(i)
    return dataframe_new_column

In [11]:
new_columns = replace_name(International_T20_Data)
new_columns

['innings',
 'data_version',
 'created',
 'revision',
 'dates',
 'gender',
 'match_type',
 'outcome.by.wickets',
 'outcome.winner',
 'overs',
 'player_of_match',
 'teams',
 'toss.decision',
 'toss.winner',
 'umpires',
 'venue',
 'city',
 'outcome.by.runs',
 'match_type_number',
 'neutral_venue',
 'outcome.method',
 'outcome.result',
 'outcome.eliminator',
 'supersubs.New Zealand',
 'supersubs.South Africa',
 'bowl_out',
 'outcome.bowl_out']

In [12]:
International_T20_Data.columns = new_columns
International_T20_Data.head()

innings  data_version  \
0  [{'1st innings': {'team': 'Australia', 'delive...           0.9   
1  [{'1st innings': {'team': 'Australia', 'delive...           0.9   
2  [{'1st innings': {'team': 'Australia', 'delive...           0.9   
3  [{'1st innings': {'team': 'Hong Kong', 'delive...           0.9   
4  [{'1st innings': {'team': 'Zimbabwe', 'deliver...           0.9   

      created  revision                         dates gender match_type  \
0  2017-02-18         2  [datetime.date(2017, 2, 17)]   male        T20   
1  2017-02-19         2  [datetime.date(2017, 2, 19)]   male        T20   
2  2017-02-23         1  [datetime.date(2017, 2, 22)]   male        T20   
3  2016-09-12         1   [datetime.date(2016, 9, 5)]   male        T20   
4  2016-06-19         1  [datetime.date(2016, 6, 18)]   male        T20   

   outcome.by.wickets outcome.winner  overs  ... outcome.by.runs  \
0                 5.0      Sri Lanka     20  ...             NaN   
1                 2.0      Sri Lanka     20  ...             NaN   
2                 NaN      Australia     20  ...            41.0   
3                 NaN      Hong Kong     20  ...            40.0   
4                 NaN       Zimbabwe     20  ...             2.0   

  match_type_number neutral_venue outcome.method outcome.result  \
0               NaN           NaN            NaN            NaN   
1               NaN           NaN            NaN            NaN   
2               NaN           NaN            NaN            NaN   
3               NaN           NaN            NaN            NaN   
4               NaN           NaN            NaN            NaN   

  outcome.eliminator supersubs.New Zealand  supersubs.South Africa  bowl_out  \
0                NaN                   NaN                     NaN       NaN   
1                NaN                   NaN                     NaN       NaN   
2                NaN                   NaN                     NaN       NaN   
3                NaN                   NaN                     NaN       NaN   
4                NaN                   NaN                     NaN       NaN   

   outcome.bowl_out  
0               NaN  
1               NaN  
2               NaN  
3               NaN  
4               NaN  

[5 rows x 27 columns]

In [13]:
International_T20_Data.isnull().sum()

innings                      0
data_version                 0
created                      0
revision                     0
dates                        0
gender                       0
match_type                   0
outcome.by.wickets         766
outcome.winner              45
overs                        0
player_of_match            162
teams                        0
toss.decision                0
toss.winner                  0
umpires                     22
venue                        0
city                       189
outcome.by.runs            696
match_type_number          733
neutral_venue             1089
outcome.method            1366
outcome.result            1372
outcome.eliminator        1401
supersubs.New Zealand     1416
supersubs.South Africa    1416
bowl_out                  1415
outcome.bowl_out          1415
dtype: int64

###As we can see some columns have 1415 null values out of 1417 entries.It will be better if we drop all these columns.

In [14]:
International_T20_Data=International_T20_Data.drop(['outcome.bowl_out','bowl_out','supersubs.South Africa','supersubs.New Zealand','outcome.eliminator','outcome.method','outcome.result'], axis = 1)

In [15]:
#lets again check our nulls again
International_T20_Data.isnull().sum()

innings                  0
data_version             0
created                  0
revision                 0
dates                    0
gender                   0
match_type               0
outcome.by.wickets     766
outcome.winner          45
overs                    0
player_of_match        162
teams                    0
toss.decision            0
toss.winner              0
umpires                 22
venue                    0
city                   189
outcome.by.runs        696
match_type_number      733
neutral_venue         1089
dtype: int64

In [16]:
#lets fill all the null valeues with 0
International_T20_Data=International_T20_Data.fillna(0)

In [17]:
#lets again check our nulls again
International_T20_Data.isnull().sum()

innings               0
data_version          0
created               0
revision              0
dates                 0
gender                0
match_type            0
outcome.by.wickets    0
outcome.winner        0
overs                 0
player_of_match       0
teams                 0
toss.decision         0
toss.winner           0
umpires               0
venue                 0
city                  0
outcome.by.runs       0
match_type_number     0
neutral_venue         0
dtype: int64

###We also see that there are columns like dataversion ,revision and created which is of no use for our analysis , lets drops these columns from our dataframe

In [18]:
International_T20_Data=International_T20_Data.drop(['data_version','created','revision'],axis =1)


In [19]:
International_T20_Data.columns.unique()

Index(['innings', 'dates', 'gender', 'match_type', 'outcome.by.wickets',
       'outcome.winner', 'overs', 'player_of_match', 'teams', 'toss.decision',
       'toss.winner', 'umpires', 'venue', 'city', 'outcome.by.runs',
       'match_type_number', 'neutral_venue'],
      dtype='object')

##Our date column is not in correct format .we need to change that

In [20]:
def modifying_date(dataframe):
    dates=[]
    for i in dataframe.dates:
        i = i.replace('[datetime.date(','').replace(')]','').replace(',','-').replace("['",'').replace("']",'')
        dates.append(i)
    return dates    

In [21]:
dates = modifying_date(International_T20_Data)

In [22]:
International_T20_Data['dates']=dates
International_T20_Data['dates'].astype(str)

0       2017- 2- 17
1       2017- 2- 19
2       2017- 2- 22
3        2016- 9- 5
4       2016- 6- 18
           ...     
1412     2016- 3- 4
1413     2016- 3- 6
1414     2016- 2- 3
1415     2016- 9- 6
1416     2016- 9- 9
Name: dates, Length: 1417, dtype: object

In [23]:
#checking our dataframe again
International_T20_Data.head()

innings        dates gender  \
0  [{'1st innings': {'team': 'Australia', 'delive...  2017- 2- 17   male   
1  [{'1st innings': {'team': 'Australia', 'delive...  2017- 2- 19   male   
2  [{'1st innings': {'team': 'Australia', 'delive...  2017- 2- 22   male   
3  [{'1st innings': {'team': 'Hong Kong', 'delive...   2016- 9- 5   male   
4  [{'1st innings': {'team': 'Zimbabwe', 'deliver...  2016- 6- 18   male   

  match_type  outcome.by.wickets outcome.winner  overs    player_of_match  \
0        T20                 5.0      Sri Lanka     20  ['DAS Gunaratne']   
1        T20                 2.0      Sri Lanka     20  ['DAS Gunaratne']   
2        T20                 0.0      Australia     20        ['A Zampa']   
3        T20                 0.0      Hong Kong     20                  0   
4        T20                 0.0       Zimbabwe     20   ['E Chigumbura']   

                        teams toss.decision toss.winner  \
0  ['Australia', 'Sri Lanka']         field   Sri Lanka   
1  ['Australia', 'Sri Lanka']         field   Sri Lanka   
2  ['Australia', 'Sri Lanka']         field   Sri Lanka   
3    ['Ireland', 'Hong Kong']           bat   Hong Kong   
4       ['Zimbabwe', 'India']         field       India   

                        umpires                              venue  \
0    ['MD Martell', 'P Wilson']           Melbourne Cricket Ground   
1     ['SD Fry', 'SJ Nogajski']     Simonds Stadium, South Geelong   
2    ['MD Martell', 'P Wilson']                      Adelaide Oval   
3       ['R Black', 'AJ Neill']  Bready Cricket Club, Magheramason   
4  ['TJ Matibiri', 'RB Tiffin']                 Harare Sports Club   

          city  outcome.by.runs  match_type_number  neutral_venue  
0            0              0.0                0.0            0.0  
1     Victoria              0.0                0.0            0.0  
2            0             41.0                0.0            0.0  
3  Londonderry             40.0                0.0            0.0  
4            0              2.0                0.0            0.0

##lets us fix some columns names by replacing '.' from them with '_'

In [49]:
International_T20_Data.rename(columns = {'outcome.by.wickets':'outcome_by_wickets', 'outcome.winner':'outcome_winner','toss.decision':'toss_decision','toss.winner':'toss_winner','outcome.by.runs':'outcome_by_runs'}, inplace = True)

##Lets us check the top winning team 

In [51]:
International_T20_Data['outcome_winner'].value_counts().head(10)

Australia       132
England         130
India           121
Pakistan        117
South Africa    101
New Zealand      97
West Indies      85
Sri Lanka        69
Afghanistan      51
Bangladesh       50
Name: outcome_winner, dtype: int64